packages/protocol/test/stability/grandamento.ts

```diff
@@ -396,23 +456,556 @@ contract('GrandaMento', (accounts: string[]) => {
+
+        it('refunds the same CELO amount as the original deposit', async () => {
+          await goldToken.approve(grandaMento.address, celoSellAmount, { from: alice })
+          await grandaMento.createExchangeProposal(stableTokenRegistryId, celoSellAmount, true, {
+            from: alice,
+          })
+
+          const grandaMentoBalanceBefore = await goldToken.balanceOf(grandaMento.address)
+          const aliceBalanceBefore = await goldToken.balanceOf(alice)
+          await grandaMento.cancelExchangeProposal(1, { from: alice })
+          const grandaMentoBalanceAfter = await goldToken.balanceOf(grandaMento.address)
+          const aliceBalanceAfter = await goldToken.balanceOf(alice)
+
+          assertEqualBN(grandaMentoBalanceBefore.minus(grandaMentoBalanceAfter), celoSellAmount)
+          assertEqualBN(aliceBalanceAfter.minus(aliceBalanceBefore), celoSellAmount)
+        })
+
+        it('refunds the entire balance if the amount to refund is higher than the balance', async () => {
+          const mockGoldToken = await MockGoldToken.new()
+          await registry.setAddressFor(CeloContractName.GoldToken, mockGoldToken.address)
+          await mockGoldToken.setBalanceOf(alice, celoSellAmount)
+
+          await grandaMento.createExchangeProposal(stableTokenRegistryId, celoSellAmount, true, {
+            from: alice,
+          })
+          const newGrandaMentoBalance = unit.times(40)
+          // Artificially lower the granda mento CELO balance.
+          await mockGoldToken.setBalanceOf(grandaMento.address, newGrandaMentoBalance)
+
+          const aliceBalanceBefore = await mockGoldToken.balanceOf(alice)
+          await grandaMento.cancelExchangeProposal(1, { from: alice })
+          const grandaMentoBalanceAfter = await mockGoldToken.balanceOf(grandaMento.address)
+          const aliceBalanceAfter = await mockGoldToken.balanceOf(alice)
+
+          assertEqualBN(grandaMentoBalanceAfter, 0)
+          assertEqualBN(aliceBalanceAfter.minus(aliceBalanceBefore), newGrandaMentoBalance)
+        })
+      })
+    })
+
+    it('reverts when called by a sender that is not permitted', async () => {
+      await grandaMento.createExchangeProposal(
+        stableTokenRegistryId,
+        stableTokenSellAmount,
+        false,
+        {
+          from: alice,
+        }
+      )
+      await assertRevertWithReason(
+        grandaMento.cancelExchangeProposal(1, { from: approver }),
+        'Sender cannot cancel the exchange proposal'
+      )
+    })
+
+    it('reverts when the proposalId does not exist', async () => {
+      await assertRevertWithReason(
+        grandaMento.cancelExchangeProposal(1, { from: approver }),
+        'Sender cannot cancel the exchange proposal'
+      )
+    })
+  })
+
+  describe('#executeExchangeProposal', () => {
+    const stableTokenSellAmount = unit.times(500)
+    const celoSellAmount = unit.times(100)
+    let sellCelo = false
+    beforeEach(async () => {
+      if (sellCelo) {
```

Comment:
Can this if be removed?

Reply:
Just added a comment explaining why it's needed for sellCelo and not !sellCelo. I figure it's best to not run it unless needed to keep tests more concise in what state they modify